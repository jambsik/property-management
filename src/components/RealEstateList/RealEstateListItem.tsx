import React from 'react';
import { Grid, Typography } from '@mui/material';
import { Box, styled } from '@mui/system';
import { RealEstate } from '../../models/RealEstate';
import { realEstateListItemImgStyles, realEstateListItemRootStyles, realEstateListItemTextStyles } from './RealEstateList.styles';
import { useTranslation } from 'react-i18next';
interface RealEstateListItemProps {
    item: RealEstate;
}

const RealEstateListItem = ({ item }: RealEstateListItemProps) => {
    const { t } = useTranslation();
    const Image = styled('img')``;
    const valuesToShow = ['street', 'yrsold', 'yearbuilt', 'saleprice', 'garagecars', 'kitchenabvgr', 'bedroomabvgr'];

    return (
        <Box sx={realEstateListItemRootStyles}>
            <Image sx={realEstateListItemImgStyles} alt="home-image" src="images/home.svg" />
            <Grid container rowSpacing={1} columnSpacing={{ xs: 2 }}>
                {valuesToShow.map((key: string) => (
                    <React.Fragment key={key}>
                        <Grid item xs={2}>
                            <Typography color="info" variant="body1">
                                {t(`realEstate.filters.${key}`)}:
                            </Typography>
                        </Grid>
                        <Grid item xs={2}>
                            <Typography sx={realEstateListItemTextStyles} color="secondary" variant="body1">
                                {item[key as keyof RealEstate]}
                            </Typography>
                        </Grid>
                    </React.Fragment>
                ))}
            </Grid>
        </Box>
    );
};

export default RealEstateListItem;

/*

                <Box key={key} sx={realEstateListItemBoxStyles}>
                 
                </Box>
*/
