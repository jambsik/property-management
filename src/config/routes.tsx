import { RouteProps } from 'react-router';
import Home from '../containers/Home/Home';
import RealEstate from '../containers/RealEstate/RealEstate';
import RealEstateDetail from '../containers/RealEstateDetail/RealEstateDetail';

export const APP_ROUTES_PATH = {
    base: '/',
    realEstate: '/real-estate',
    realEstateDetail: '/real-estate/:id',
};

export const getAppRoutes = (): RouteProps[] => [
    {
        path: APP_ROUTES_PATH.base,
        element: <Home />,
    },
    {
        path: APP_ROUTES_PATH.realEstate,
        element: <RealEstate />,
    },
    {
        path: APP_ROUTES_PATH.realEstateDetail,
        element: <RealEstateDetail />,
    },
];

export const getDetailPathRoute = (id: string | number) => APP_ROUTES_PATH.realEstateDetail.replace(':id', `${id}`);
