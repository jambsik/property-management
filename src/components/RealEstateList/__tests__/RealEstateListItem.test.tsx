import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import RealEstateListItem from '../RealEstateListItem';
import { addTestProviders } from '../../../__tests__/__utils__/renderHelper';
import { basicRealEstateItemDataMock } from '../../../__tests__/__fixtures__/realEstateDataMock';
import { getDetailPathRoute } from '../../../config/routes';

describe('Real state item list test', () => {
    const pageObject = {
        root: `list-item-${basicRealEstateItemDataMock.id}`,
    };

    it('Should render component without errors', () => {
        const { container } = render(addTestProviders(<RealEstateListItem item={basicRealEstateItemDataMock} />));
        expect(container).toMatchSnapshot();
    });

    it('Should render item content', () => {
        const { queryByText } = render(addTestProviders(<RealEstateListItem item={basicRealEstateItemDataMock} />));

        const streetElement = queryByText(basicRealEstateItemDataMock.street);
        const yearBuiltElement = queryByText(basicRealEstateItemDataMock.yearbuilt);
        const salePriceElement = queryByText(basicRealEstateItemDataMock.saleprice);

        expect(streetElement).toBeInTheDocument();
        expect(yearBuiltElement).toBeInTheDocument();
        expect(salePriceElement).toBeInTheDocument();

        expect(streetElement).toHaveTextContent(`${basicRealEstateItemDataMock.street}`);
        expect(yearBuiltElement).toHaveTextContent(`${basicRealEstateItemDataMock.yearbuilt}`);
        expect(salePriceElement).toHaveTextContent(`${basicRealEstateItemDataMock.saleprice}`);
    });

    it('Should navigate to detail page after click on item', () => {
        expect(window.location.href).toEqual('http://localhost/');

        const { getByTestId } = render(addTestProviders(<RealEstateListItem item={basicRealEstateItemDataMock} />));

        fireEvent.click(getByTestId(pageObject.root));

        expect(window.location.href).toEqual(`http://localhost${getDetailPathRoute(basicRealEstateItemDataMock.id)}`);
    });
});
