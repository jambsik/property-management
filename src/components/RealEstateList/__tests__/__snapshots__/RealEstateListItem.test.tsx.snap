// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Real state item list test Should render component without errors 1`] = `
<div>
  <a
    href="/real-estate/1"
  >
    <div
      class="MuiBox-root css-759u60"
      data-testid="list-item-1"
    >
      <img
        alt="home-image"
        class="css-1ninaiz"
        src="/images/home.svg"
      />
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined css-4gcdpv-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.street
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            Pave
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.yrsold
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            2008
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.yearbuilt
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            2003
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.saleprice
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            208500
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.garagecars
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            2
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.kitchenabvgr
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            1
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-7g5z04-MuiTypography-root"
          >
            realEstate.filters.bedroomabvgr
            :
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-a0vo9r-MuiGrid-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-1wlo5s9-MuiTypography-root"
          >
            3
          </p>
        </div>
      </div>
    </div>
  </a>
</div>
`;
